src/zod-to-json.ts

```diff
@@ -45,17 +46,23 @@ export const zodSchemaToJson: (
   registry: z.core.$ZodRegistry<{ id?: string }>,
   io: 'input' | 'output',
 ) => ReturnType<typeof deleteInvalidProperties> = (zodSchema, registry, io) => {
-  const result = z.toJSONSchema(zodSchema, {
-    io,
+  const jsonSchemaGenerator = new JSONSchemaGenerator({
```

Comment:
To be sure that is a public API and the correct method I opened an issue on zod:
- https://github.com/colinhacks/zod/issues/4749